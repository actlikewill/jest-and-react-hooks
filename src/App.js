import React from 'react';
import Header from './Header'
import Clicker from './Clickers'
import Fetch from './Fetch'
import './App.css';

const loadFbLoginApi = () => {
  window.fbAsyncInit = function() {
      window.FB.init({
          appId      : process.env.REACT_APP_FACEBOOK_APP_ID,
          cookie     : true,  
          xfbml      : true, 
          version    : 'v2.5'
      });
  };
  (function(d, s, id) {
      var js, fjs = d.getElementsByTagName(s)[0];
      if (d.getElementById(id)) return;
      js = d.createElement(s); js.id = id;
      js.src = "//connect.facebook.net/en_US/sdk.js";
      fjs.parentNode.insertBefore(js, fjs);
  }(document, 'script', 'facebook-jssdk'));
}



function App() {
  loadFbLoginApi()
  return (
    <div className="App">
      <Header text='Hello World' />
      <Clicker />
      <Fetch url="https://jsonplaceholder.typicode.com/todos/1" />
    </div>
  );
}

export default App;
