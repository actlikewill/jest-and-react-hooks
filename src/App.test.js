import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';

describe('renders the app', () => {
  beforeAll(() => {
    let scriptTag = document.createElement('script');
    document.body.appendChild(scriptTag)
  })
  afterEach(cleanup)
  it('should take a snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot();
  });  
})


