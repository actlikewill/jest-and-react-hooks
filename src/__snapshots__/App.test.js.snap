// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header>
      <h1
        class="headerStyle"
        data-testid="header-tag"
      >
        Hello World
      </h1>
    </header>
    <div>
      <h1
        data-testid="count"
      >
        0
      </h1>
      <button
        data-testid="increase"
      >
        Increase
      </button>
      <button
        data-testid="decrease"
      >
        Decrease
      </button>
    </div>
    <span
      data-testid="loading"
    >
      Loading data ...
    </span>
  </div>
</DocumentFragment>
`;
